middleware/http_cluster_test.go

```diff
@@ -0,0 +1,307 @@
+					}
+				}
+				if testCase.expectedLogs != "" {
+					require.True(t, bytes.Contains(buf.Bytes(), []byte(fmt.Sprintf(testCase.expectedLogs, softValidation))))
+				}
+			})
+		}
+	}
+}
+
+func TestClusterValidationMiddlewareWithExcludedPaths(t *testing.T) {
+	testCases := map[string]struct {
+		softValidation       bool
+		requestPath          string
+		excludedPaths        []string
+		expectedStatusCode   int
+		expectedErrorMessage string
+	}{
+		"when soft validation is enabled and request path is excluded no error is returned": {
+			softValidation:       true,
+			requestPath:          "Test/Me",
+			excludedPaths:        []string{"Exclude/Me", "Test/Me", "Do/Not/Test/Me"},
+			expectedStatusCode:   http.StatusOK,
+			expectedErrorMessage: "",
+		},
+		"when soft validation is disabled and request path is excluded no error is returned": {
+			softValidation:       false,
+			requestPath:          "Test/Me",
+			excludedPaths:        []string{"Exclude/Me", "Test/Me", "Do/Not/Test/Me"},
+			expectedStatusCode:   http.StatusOK,
+			expectedErrorMessage: "",
+		},
+		"when soft validation is disabled and request path is implicitly excluded no error is returned": {
+			softValidation:       false,
+			requestPath:          "metrics",
+			excludedPaths:        []string{"Exclude/Me", "Test/Me", "Do/Not/Test/Me"},
+			expectedStatusCode:   http.StatusOK,
+			expectedErrorMessage: "",
+		},
+		"when soft validation is enabled and request path is not excluded no error is returned": {
+			softValidation:       true,
+			requestPath:          "Test/Me",
+			excludedPaths:        []string{"Exclude/Me", "Do/Not/Test/Me"},
+			expectedStatusCode:   http.StatusOK,
+			expectedErrorMessage: "",
+		},
+		"when soft validation is disabled and request path is not excluded an error is returned": {
+			softValidation:       false,
+			requestPath:          "Test/Me",
+			excludedPaths:        []string{"Exclude/Me", "Do/Not/Test/Me"},
+			expectedStatusCode:   http.StatusNetworkAuthenticationRequired,
+			expectedErrorMessage: `rejected request with wrong cluster validation label "client-cluster" - it should be "server-cluster"`,
+		},
+	}
+	for testName, testCase := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			m := ClusterValidationMiddleware("server-cluster", testCase.excludedPaths, testCase.softValidation, log.NewNopLogger())
+			handler := Merge(m).Wrap(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusOK)
+			}))
+
+			recorder := httptest.NewRecorder()
+			req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost:8080/%s", testCase.requestPath), nil)
+			require.NoError(t, err)
+			req.Header[clusterutil.ClusterVerificationLabelHeader] = []string{"client-cluster"}
+
+			handler.ServeHTTP(recorder, req)
+			require.Equal(t, testCase.expectedStatusCode, recorder.Code)
+			if recorder.Code != http.StatusOK {
+				var clusterValidationErr clusterValidationError
```

Comment:
Could you also test that the server responds with `Content-Type` `application/json`?